libs/dh/message-archive/feature-log-search/src/lib/dh-message-archive-log-search.component.ts

```diff
@@ -0,0 +1,211 @@
+  WattIconModule,
+  WattInputModule,
+  WattCheckboxModule,
+  WattBadgeModule,
+} from '@energinet-datahub/watt';
+import {
+  DhMessageArchiveDataAccessApiModule,
+  DhMessageArchiveDataAccessBlobApiModule,
+} from '@energinet-datahub/dh/message-archive/data-access-api';
+import {
+  SearchCriteria,
+  SearchResultItemDto,
+} from '@energinet-datahub/dh/shared/domain';
+import { LetModule } from '@rx-angular/template';
+import { TranslocoModule } from '@ngneat/transloco';
+import { DhMessageArchiveLogSearchResultScam } from './searchresult/dh-message-archive-log-search-result.component';
+import { BusinessReasonCodes } from '../../../models/businessReasonCodes';
+import { DocumentTypes } from '../../../models/documentTypes';
+import { RoleTypes } from '../../../models/roleTypes';
+import { Subject } from 'rxjs';
+import { ActivatedRoute, Router } from '@angular/router';
+
+@Component({
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  selector: 'dh-message-archive-log-search',
+  styleUrls: ['./dh-message-archive-log-search.component.scss'],
+  templateUrl: './dh-message-archive-log-search.component.html',
+  providers: [
+    DhMessageArchiveDataAccessApiModule,
+    DhMessageArchiveDataAccessBlobApiModule,
+  ],
+})
+export class DhMessageArchiveLogSearchComponent implements OnDestroy {
+  private destroy$ = new Subject<void>();
+  private regexBlobName = new RegExp(
+    /\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01])\/.*/
+  );
+
+  searchResult$ = this.store.searchResult$;
+  searching$ = this.store.isSearching$;
+  hasSearchError$ = this.store.hasGeneralError$;
+  continuationToken$ = this.store.continuationToken$;
+
+  searchResultsDtos: Array<SearchResultItemDto> = [];
+  businessReasonCodes = BusinessReasonCodes;
+  searching = false;
+  documentTypes = DocumentTypes;
+  roleTypes = RoleTypes;
+  pageSizes = [250, 500, 750, 1000];
+  pageNumber = 1;
+
+  private initDateFrom = (): Date => {
+    const from = new Date();
+    from.setUTCMonth(new Date().getMonth() - 1);
+    from.setUTCHours(0, 0, 0);
+    return from;
+  };
+  private initDateTo = (): Date => {
+    const to = new Date();
+    to.setUTCHours(23, 59, 59);
+    return to;
+  };
+
+  // eslint-disable-next-line @typescript-eslint/member-ordering
+  searchCriteria: SearchCriteria = {
+    messageId: null,
+    reasonCode: null,
+    dateTimeFrom: this.initDateFrom().toISOString().split('.')[0] + 'Z',
+    dateTimeTo: this.initDateTo().toISOString().split('.')[0] + 'Z',
+    senderRoleType: null,
+    rsmName: null,
+    includeRelated: false,
+    traceId: null,
+    functionName: null,
+    invocationId: null,
+    maxItemCount: 250,
+  };
+
+  constructor(
+    private store: DhMessageArchiveDataAccessApiModule,
+    private router: Router,
+    private currentRoute: ActivatedRoute,
+    private logStore: DhMessageArchiveDataAccessBlobApiModule
+  ) {
+    this.store.searchResult$.subscribe((searchResult) => {
+      this.searchResultsDtos = searchResult;
```

Comment:
Suggestion: since this is only used in the template, consider removing this and use the `async` pipe instead (same goes for the other subscriptions where it's possible).